porcupine/pluginloader.py

```diff
@@ -183,6 +183,16 @@ def _run_setup_and_set_status(info: PluginInfo) -> None:
     handler.emit = error_log.append  # type: ignore
     logger.addHandler(handler)
 
+    if not hasattr(info.module, "setup"):
+        info.status = Status.SETUP_FAILED
+        info.error = (
+            "There is no setup() function. Make sure to include a setup function into your"
+            " plugin.\nTo learn more about Porcupine's plugin API, visit"
+            " https://akuli.github.io/porcupine/plugin-intro.html"
+        )
+        log.warning(f"Calling {info.name!r} plugin's setup() function failed.\n{info.error}")
+        return
```

Comment:
This early-return skips code that removes a logging handler. It would be better to remove it, even though a plugin without a `setup()` likely won't log much.